rules/legacy_xcodeproj.bzl

```diff
@@ -463,6 +464,31 @@ xcodeproj_lldbinit = rule(
     doc = "Internal testing rule relying on assumptions about the xcodeproj rule above",
 )
 
+def _xcodeproj_project_options_impl(ctx):
+    # Map of the attr name to the options value expected by XcodeGen.
+    options_map = {
+        "uses_tabs": "usesTabs",
+        "indent_width": "indentWidth",
+        "tab_width": "tabWidth",
+    }
+
+    # Accumulate all provided options into a Starlark dict.
+    options = {}
+    for attr, option in options_map.items():
+        if hasattr(ctx.attr, attr):
+            options[option] = getattr(ctx.attr, attr)
+    return [_ProjectOptionsInfo(options = options)]
+
+xcodeproj_project_options = rule(
+    implementation = _xcodeproj_project_options_impl,
+    attrs = {
+        "uses_tabs": attr.bool(mandatory = False),
+        "indent_width": attr.int(mandatory = False),
+        "tab_width": attr.int(mandatory = False),
+    },
+    doc = "Rule to change project options",
```

Comment:
Minor: maybe add a note to cover what happens if you _don't_ provide this option: e.g. if you don't set it then it will fall back to xcode's defaults ( assuming xcodegen doesn't have their own flavor of default value )

Reply:
```suggestion
    doc = "Rule to change project options, values will fallback to Xcode default if not provided",
```